api/src/org/labkey/api/data/ExpDataFileConverter.java

```diff
@@ -129,7 +129,15 @@ else if (dataObject.has(ExperimentJSONConverter.DATA_FILE_URL) && dataObject.get
                 {
                     // Check for file at file root
                     String root = FileContentService.get().getFileRootPath(container, FileContentService.ContentType.files).toString();
-                    data = expSvc.getExpDataByURL(FileUtil.createUri("/" + root + dataFileURL).toString(), container);
+
+                    // non-unix needs to prepend with a slash
+                    if (!root.startsWith("/") && !root.startsWith("\\"))
+                    {
+                        root = "/" + root;
+                    }
+
+                    String path = FileUtil.createUri(root + dataFileURL).toString();
+                    data = expSvc.getExpDataByURL(path, container);

```

Comment:
This was the change I mentioned earlier (requires `java.net.URI` to be imported as well)
```suggestion
                    java.nio.file.Path root = FileContentService.get().getFileRootPath(container, FileContentService.ContentType.files);
                    URI dataFileUri = root.resolve(dataFileURL).toUri();
                    data = expSvc.getExpDataByURL(dataFileUri.toString(), container);
```

Reply:
This didn't work for me at least on Windows.  Tried several different approaches to get the same result for Linux and Windows but only way that worked was string manipulation.